src/v1/entity/utils/validateKeyInput.unit.test.ts

```diff
@@ -0,0 +1,154 @@
+  validateKeyInput,
+  UnrecognizedKeyInputPropertyError,
+  InvalidKeyInputValueTypeError
+} from './validateKeyInput'
+
+describe('validateKeyInput', () => {
+  const path = 'some.path'
+
+  it('always validates key any', () => {
+    const input = ['any', { value: 'that you want' }]
+
+    expect(() => validateKeyInput(any(), input, path)).toThrow(
+      new UnrecognizedKeyInputPropertyError({ path })
+    )
+
+    expect(() => validateKeyInput(any().key(), input, path)).not.toThrow()
+  })
+
+  it('validates valid key binaries', () => {
+    const validInput = Buffer.from('toto')
+
+    expect(() => validateKeyInput(binary(), validInput, path)).toThrow(
+      new UnrecognizedKeyInputPropertyError({ path })
+    )
+
+    const keyBin = binary().key()
+    expect(() => validateKeyInput(keyBin, validInput, path)).not.toThrow()
+
+    const invalidInput = 'some string'
+    // @ts-expect-error
+    expect(() => validateKeyInput(keyBin, invalidInput, path)).toThrow(
+      new InvalidKeyInputValueTypeError({ expectedType: 'binary', keyInput: invalidInput, path })
+    )
+  })
+
+  it('validates valid key booleans', () => {
+    const validInput = true
+
+    expect(() => validateKeyInput(boolean(), validInput, path)).toThrow(
+      new UnrecognizedKeyInputPropertyError({ path })
+    )
+
+    const keyBool = boolean().key()
+    expect(() => validateKeyInput(keyBool, validInput, path)).not.toThrow()
+
+    const invalidInput = 'some string'
+    // @ts-expect-error
+    expect(() => validateKeyInput(keyBool, invalidInput, path)).toThrow(
+      new InvalidKeyInputValueTypeError({ expectedType: 'boolean', keyInput: invalidInput, path })
+    )
+  })
+
+  it('validates valid key numbers', () => {
+    const validInput = 42
+
+    expect(() => validateKeyInput(number(), validInput, path)).toThrow(
+      new UnrecognizedKeyInputPropertyError({ path })
+    )
+
+    const keyNum = number().key()
+    expect(() => validateKeyInput(keyNum, validInput, path)).not.toThrow()
+
+    const invalidInput = 'some string'
+    // @ts-expect-error
+    expect(() => validateKeyInput(keyNum, invalidInput, path)).toThrow(
+      new InvalidKeyInputValueTypeError({ expectedType: 'number', keyInput: invalidInput, path })
+    )
+  })
+
+  it('validates valid key strings', () => {
+    const validInput = 'foo'
+
+    expect(() => validateKeyInput(string(), validInput, path)).toThrow(
+      new UnrecognizedKeyInputPropertyError({ path })
+    )
+
+    const keyStr = string().key()
+    expect(() => validateKeyInput(keyStr, validInput, path)).not.toThrow()
+
+    const invalidInput = 42
+    // @ts-expect-error
+    expect(() => validateKeyInput(keyStr, invalidInput, path)).toThrow(
+      new InvalidKeyInputValueTypeError({ expectedType: 'string', keyInput: invalidInput, path })
+    )
+  })
+
+  it('validates valid key lists', () => {
+    const validInput = ['foo']
+
+    expect(() => validateKeyInput(list(string().required()), validInput, path)).toThrow(
```

Comment:
I think we should adhere to the single assertion per test concept.
I know it's not something we've really done, but it helps debugging issues more easily and increases readability.

wdyt?

Reply:
Agreed

Reply:
Agreed